compiler/dialects/acc.py

```diff
@@ -189,12 +212,86 @@ def verify_(self) -> None:
             )
 
 
+class AcceleratorSymbolOpTrait(SymbolOpInterface):
+    def get_sym_attr_name(self, op: Operation) -> StringAttr | None:
+        assert isinstance(op, AcceleratorOp)
+        return StringAttr(op.name_prop.string_value())
+
+
+@irdl_op_definition
+class AcceleratorOp(IRDLOperation):
+    """
+    Declares an accelerator that can be configures, launched, etc.
+    `fields` is a dictionary mapping accelerator configuration names to
+    CSR addresses.
+    """
+
+    name = "acc2.accelerator"
+
+    traits = frozenset([AcceleratorSymbolOpTrait()])
+
+    name_prop = prop_def(SymbolRefAttr, prop_name="name")
+
+    fields = prop_def(DictionaryAttr)
+
+    launch_addr = prop_def(IntegerAttr)
+
+    barrier_sw_barrier = prop_def(
+        IntegerAttr
+    )  # TODO: this will be reworked in a later version
```

Comment:
This attribute now has a duplicate "barrier" in the name,

Reply:
Yeah that's a bit silly, I propose we make the barrier enable (if required for later editions, an accelerator parameter instead, then it will also be set just once)